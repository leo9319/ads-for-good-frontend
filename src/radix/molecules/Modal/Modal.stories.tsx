import type { Meta, StoryObj } from '@storybook/react';
import Modal from '@radix-styles/molecules/Modal';
import React from 'react';

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: 'Library/Molecules/Modal',
  component: Modal,
  parameters: {
    // Optional parameter to center the component in the Canvas. More info: https://storybook.js.org/docs/configure/story-layout
    layout: 'centered',
  },
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ['autodocs'],
  args: {
    className: '',
    children: <div>Default Child</div>,
  },
} satisfies Meta<typeof Modal>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Default: Story = {
  args: {
    children: (
      <div style={{ padding: '1rem' }}>
        <h1>Filter</h1>
        <ul>
          <li>Option 1</li>
          <li>Option 2</li>
          <li>Option 3</li>
        </ul>
      </div>
    ),
    buttonName: 'Filter',
  },
};
