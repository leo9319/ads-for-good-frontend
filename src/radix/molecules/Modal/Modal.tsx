import React, { ReactElement } from 'react';

import Button from '@radix-styles/atoms/Button';
import * as Dialog from '@radix-ui/react-dialog';
import { classNames } from '@utils/common/classNames';
import { VisuallyHidden } from '@radix-ui/themes';

import styles from './Modal.module.scss';

export interface ModalProps {
  /**
   * Button name which will that open up the modal/popup
   * @property
   */
  buttonName: string;
  /**
   * Styling for button
   * @property
   */
  className?: string;
  /**
   * Children as jsx, react component will show inside the modal
   * @property
   */
  children: React.ReactNode;
  /**
   * Boolean state to represent hide/show of the modal
   * @property
   * @default false
   */
  open?: boolean;
  /**
   * toggle function to change between open and close state of the modal
   * @property
   */
  setOpen?: React.Dispatch<React.SetStateAction<boolean>>;
}

/**
 * Modal component is to show the content in a popup/modal.
 *
 * @group Molecules
 * @category Radix Component
 *
 * @param {ModalProps} props - The Modal props
 * @returns {ReactElement} - Modal Component
 */
export const Modal = ({
  open,
  setOpen,
  buttonName,
  className,
  children,
}: ModalProps): ReactElement => {
  return (
    <Dialog.Root open={open} onOpenChange={setOpen}>
      <Dialog.Trigger asChild>
        <Button
          size="sm"
          aria-haspopup="dialog"
          aria-controls={open ? 'dialog-content' : undefined}
          mode="transparent"
          text={buttonName}
          className={classNames(className)}
        />
      </Dialog.Trigger>
      <Dialog.Portal>
        <Dialog.Overlay className={styles.Overlay} />
        <Dialog.Content className={styles.Content} id="dialog-content">
          <VisuallyHidden>
            <Dialog.DialogTitle>Accessible Dialog Title</Dialog.DialogTitle>
          </VisuallyHidden>
          {children}
        </Dialog.Content>
      </Dialog.Portal>
    </Dialog.Root>
  );
};
export default Modal;
