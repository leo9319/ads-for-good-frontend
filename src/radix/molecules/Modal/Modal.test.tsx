import React from 'react';

import { render } from '@testing-library/react';
import Modal from './Modal';

describe('Modal', () => {
  it('should render successfully', () => {
    const { baseElement } = render(
      <Modal buttonName="Filter Button">
        <>
          <h1>Title</h1>
          <p>Content</p>
        </>
      </Modal>
    );
    expect(baseElement).toMatchSnapshot();
  });
});
