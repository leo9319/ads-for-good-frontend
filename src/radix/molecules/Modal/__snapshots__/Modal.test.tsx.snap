// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal should render successfully 1`] = `
<body>
  <div>
    <button
      aria-expanded="false"
      aria-haspopup="dialog"
      aria-label="Filter Button"
      class="rt-reset rt-BaseButton rt-r-size-2 rt-variant-solid rt-Button wvButton sm transparent none"
      data-accent-color=""
      data-state="closed"
      type="button"
    >
      <div
        class="buttonContent textOnly"
      >
        <span>
          Filter Button
        </span>
      </div>
    </button>
  </div>
</body>
`;
